src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -8411,6 +8411,112 @@ public static int Main()
 
         #endregion
 
+        #region SkipLocalsInitAttribute
+
+        [Fact]
+        public void SkipLocalsInitAttributeRecognizedAsWellKnownTypeWhenDeclaredInSource()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GlobalNamespace
+                .GetNestedNamespace("System")
+                .GetNestedNamespace("Runtime")
+                .GetNestedNamespace("CompilerServices")
+                .GetTypeMember("SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeRecognizedAsWellKnownTypeWhenDeclaredInMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation("", references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GlobalNamespace
+                .GetNestedNamespace("System")
+                .GetNestedNamespace("Runtime")
+                .GetNestedNamespace("CompilerServices")
+                .GetTypeMember("SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeFromSourceRecognizedAsWellKnownTypeWhenDeclaredInSourceAndMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation(source, references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+
+            Assert.Equal(wellKnownType.GetType().Name, "SourceNamedTypeSymbol");
```

Comment:
Perhaps `Assert.IsType<SourceNamedTypeSymbol>(wellKnownType)` or `Assert.Equal(wellKnownType.ContainingModule, comp.SourceModule)`.

Reply:
Yes, I like the first suggestion.